### PR TITLE
Expose smart pointers

### PR DESCRIPTION
**Pull request by ajyoon**
In the process of migrating from the older opencv-rust (at opencv 2.4), I found that the internal pointers of smart pointers had been hidden from the docs (in the case of reads) and no longer settable. This is functionality I need for some evil pointer hacks in
my project.

This PR restores this functionality by:
- Removing the `#[doc(hidden)]` annotation from the generated `as_raw_*` methods, so users can more easily discover their existence. I don't see any benefit to hiding these methods; we're all responsible adults here.
- Adding `unsafe fn from_raw_ptr()` to smart pointer types.

This follows the pattern of rust stdlib practices for dealing with smart pointers - for example, `Vec` has [`from_raw_parts`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts), which is rightly marked as unsafe. Similarly, I believe we should be able to unsafely construct opencv objects from raw pointers.

### CODE REVIEW

**Comment by ajyoon** [comment]
PS - thanks @twistedfall for taking on this project recently and getting us to opencv3, and thanks @kali for making it happen from the beginning! This crate is a life saver!

File: gen_rust.py [hunk 1]
```diff
@@ -406,6 +406,11 @@ def write_exc(filename, action):
 
     ### dnn ###
     "cv::dnn::_Range",
+
+    ### xfeatures2d ###
+    # Requires opencv to be configured with -DOPENCV_ENABLE_NONFREE=ON
+    # and needs some gen_rust.py changes to build correctly.
+    "cv::xfeatures2d",
```
Comment (ajyoon) [inline gen_rust.py]: I happen to have opencv compiled with `-DOPENCV_ENABLE_NONFREE=ON`, which causes this new header to appear which isn't currently in the committed rs files. It actually compiled fine on my machine, but travis doesn't like it. In any case, I think adding those new classes should be a separate PR.

Comment (ajyoon) [reply gen_rust.py]: moved to https://github.com/twistedfall/opencv-rust/pull/37/files#diff-a7b0a2dee0126cddf994326e705a91eaR121

File: build.rs [hunk 2]
```diff
@@ -118,6 +118,7 @@ fn build_wrapper(opencv: pkg_config::Library) {
         "text",
         "tracking",
         "viz", // includes platform-specific headers like /usr/include/x86_64-pc-linux-gnu/opencv2/viz.hpp
+        "xfeatures2d", // only appears in some builds, maybe platform or opencv compile flag specific
```
Comment (ajyoon) [inline build.rs:121]: for future explorers who may want to reintegrate this, my opencv was compiled with:

opencv_contrib 3.4

opencv 3.4.6 using flags:
```sh
cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_PERF_TESTS=OFF \
    -DBUILD_TESTS=OFF \
    -DINSTALL_TESTS=OFF \
    -DBUILD_DOCS=OFF \
    -DBUILD_EXAMPLES=OFF \
    -DBUILD_opencv_apps= \
    -DWITH_IPP=OFF \
    -DPYTHON_EXECUTABLE=OFF \
    -DINSTALL_PYTHON_EXAMPLES=OFF \
    -DWITH_LAPACK=ON \
    -DWITH_EIGEN=ON \
    -DBUILD_SHARED_LIBS=ON \
    -DWITH_TBB=ON \
    -DOPENCV_ENABLE_NONFREE=ON \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DOPENCV_EXTRA_MODULES_PATH=~/tools/opencv_contrib/modules \
    ..
```

on arch linux

**Comment by kali** [comment]
> PS - thanks @twistedfall for taking on this project recently and getting us to opencv3, and thanks @kali for making it happen from the beginning! This crate is a life saver!

Happy to see it was not in vain !
